changed documentation theme to pydata_sphinx_theme

DON-BRAN opened the pull request:
Implementation of the suggestion here #1217

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

joshmoore commented:
Wow! That's quite the update.

![Screen Shot 2022-11-02 at 19 36 02](https://user-images.githubusercontent.com/88113/199573585-111dfa61-9da9-4b74-b4a8-febb3c3eb402.png)

jakirkham commented:
cc @MSanKeys963

MSanKeys963 commented:
Hi @DON-BRAN. Thanks for sending the PR.

> Wow! That's quite the update.
Indeed, @joshmoore.

@DON-BRAN, I was wondering if there's a possibility to restructure the homepage of Zarr documentation to something like [this](https://pandas.pydata.org/docs/index.html)?

Seems like most of the projects using `pydata-sphinx-theme` are using a similar format for their homepage.

DON-BRAN commented:
> Hi @DON-BRAN. Thanks for sending the PR. The change looks good.
> 
> > Wow! That's quite the update.
> 
> Indeed, @joshmoore.
> 
> @DON-BRAN, I was wondering if there's a possibility to restructure the homepage of Zarr documentation to something like [this](https://pandas.pydata.org/docs/index.html)?
> 
> Seems like most of the OSS [projects](https://pydata-sphinx-theme.readthedocs.io/en/stable/examples/gallery.html) using `pydata-sphinx-theme` are using a similar format for their homepage.

Hello @MSanKeys963, i'll be glad to work on that. I will begin looking into it.
I'll drop any questions I have under this PR thread

rabernat commented:
Folks looking to preview this site can go straight to https://zarr--1242.org.readthedocs.build/en/1242/

My overall impression is that this is a great change. I support migrating to pydata_sphinx_theme. 🎉 

We should probably do a thorough line-be-line comparison of old vs new site  to make sure all formatting and other features make it through cleanly.

Mubarraqqq commented:
Alright @rabernat

MSanKeys963 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,139 @@
+:user:`Josh Moore <joshmoore>`
+:user:`Juan Nunez-Iglesias <jni>`
+:user:`Justin Swaney <jmswaney>`
+:user:`Mads R. B. Kristensen <madsbk>`
+:user:`Mamy Ratsimbazafy <mratsim>`
+:user:`Martin Durant <martindurant>`
+:user:`Matthew Rocklin <mrocklin>`
+:user:`Matthias Bussonnier <Carreau>`
+:user:`Mattia Almansi <malmans2>`
+:user:`Noah D Brenowitz <nbren12>`
+:user:`Oren Watson <orenwatson>`
+:user:`Pavithra Eswaramoorthy <pavithraes>`
+:user:`Poruri Sai Rahul <rahulporuri>`
+:user:`Prakhar Goel <newt0311>`
+:user:`Raphael Dussin <raphaeldussin>`
+:user:`Ray Bell <raybellwaves>`
+:user:`Richard Scott <RichardScottOZ>`
+:user:`Richard Shaw <jrs65>`
+:user:`Ryan Abernathey <rabernat>`
+:user:`Ryan Williams <ryan-williams>`
+:user:`Saransh Chopra <Saransh-cpp>`
+:user:`Sebastian Grill <yetyetanotherusername>`
+:user:`Shikhar Goenka <shikharsg>`
+:user:`Shivank Chaudhary <Alt-Shivam>`
+:user:`Stephan Hoyer <shoyer>`
+:user:`Stephan Saalfeld <axtimwalde>`
+:user:`Tarik Onalan <onalant>`
+:user:`Tim Crone <tjcrone>`
+:user:`Tobias Kölling <d70-t>`
+:user:`Tom Augspurger <TomAugspurger>`
+:user:`Tom White <tomwhite>`
+:user:`Tommy Tran <potter420>`
+:user:`Trevor Manz <manzt>`
+:user:`Vincent Schut <vincentschut>`
+:user:`Vyas Ramasubramani <vyasr>`
+:user:`Zain Patel <mzjp2>`
+:user:`gsakkis`
+:user:`hailiangzhang <hailiangzhang>`
+:user:`pmav99 <pmav99>`
+:user:`sbalmer <sbalmer>`
```
Hi @DON-BRAN. Currently, the `Acknowledgements` are shown like this. 👇🏻
I think this is not reader-friendly. Do you think we can have a bulleted list of items?

<img width="695" alt="Screenshot 2022-12-13 at 00 24 57" src="https://user-images.githubusercontent.com/20305658/207130376-4170e37a-559f-47ee-936e-451f27608bba.png">

MSanKeys963 commented:
Hi @DON-BRAN. Thanks for working on the change.
I've gone through it and added some comments above. Let me know what you think.

DON-BRAN commented:
> Hi @DON-BRAN. Thanks for working on the change. I've gone through it and added some comments above. Let me know what you think.
> 
> Also, can we add a Twitter icon with a hyperlink at the top right?

Hi @MSanKeys963, thank you for the review. I will make the necessary changes. 
About the Twitter Icon I believe it's possible. 
Please can you provide the twitter handle so I can integrate 🙂

joshmoore commented:
@zarr-developers/python-core-devs, this is moving toward completion and there don't look to be any down votes (only 4 likes on [twitter](https://twitter.com/zarr_dev/status/1593503660553781248) though ;) )

One thing to be aware of: ⚠️ the URLs for the landing page (e.g. https://zarr.readthedocs.io/en/stable/#installation) have changed (e.g. https://zarr--1242.org.readthedocs.build/en/1242/installation.html)

joshmoore commented:
@normanrz suggested client-side javascript for the redirects like https://stackoverflow.com/a/21198129/56887

DON-BRAN commented:
> @normanrz suggested client-side javascript for the redirects like https://stackoverflow.com/a/21198129/56887

Hi @joshmoore, thank you for the review. I made some updates with regards to your review. The URL https://zarr--1242.org.readthedocs.build/en/1242/#installation now redirects to https://zarr--1242.org.readthedocs.build/en/1242/installation.html.
I'll be happy to make any further changes if necessary 🙂

joshmoore commented:
Kudos, @DON-BRAN (and @normanrz). Could you add all the other redirects and then I'd be a very happy camper ⛺

joshmoore commented:
@DON-BRAN: do you see getting to the other redirects?

joshmoore commented on docs/index.rst:
```diff
@@ -1,142 +1,102 @@
+.. _zarr_docs_mainpage:
+
+******************
+Zarr documentation
```
Should this perhaps be `Zarr-python`?

MSanKeys963 replied:
+1 for `Zarr-Python`.

DON-BRAN replied:
+1 for Zarr-Python

joshmoore commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,139 @@
+If you are using Zarr, we would `love to hear about it
+<https://github.com/zarr-developers/community/issues/19>`_.
+
+Acknowledgments
```
Not sure acknowledgements belongs under "contributing"

MSanKeys963 replied:
Yes, I agree. Can we have a separate page for Acknowledgements, @ DON-BRAN?

DON-BRAN replied:
Yes, I do agree with Acknowledgements being on a separate, it is way better and stands out in a nice way

joshmoore commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,139 @@
+* :user:`pmav99 <pmav99>`
+* :user:`sbalmer <sbalmer>`
+
+Contents
```
Not sure "Contents" and below belongs under "Getting started"

MSanKeys963 replied:
I think we should remove `Tutorials`, `API Reference`, `Specifications` and `Release Notes` from 'Contents' as they're already covered at the top.

We can leave `Contributing to Zarr` on 'Getting Started' if we want.

Also, maybe this is for me only, but I'm getting `404` for all of the various Tutorials and Release Notes when accessing from [here](https://zarr--1242.org.readthedocs.build/en/1242/getting_started.html).

Can you please check @DON-BRAN and @joshmoore?

DON-BRAN replied:
Yes, there's not really any need for the reiteration under `Getting started` though it could still be left so, but avoiding it looking bulky and getting it more organized, i suggest they could be taken out

@MSanKeys963 Everything opens fine for me, and looking really great

MSanKeys963 replied:
> @MSanKeys963 Everything opens fine for me, and looking really great

👍🏻 

Can you please remove the sections mentioned above and add a separate page for acknowledgements?

joshmoore reviewed (commented):
Added a few more comments from looking at https://zarr--1242.org.readthedocs.build/en/1242/ closely while looking for redirects. We could potentially handle these in follow-ups though.

One way or the other, it's a huge improvement, thanks @DON-BRAN!

DON-BRAN commented:
Hello @joshmoore @MSanKeys963 , I made updates as suggested above here are some snapshots of my changes

- I added the a link to the `zar-dev` website to the top of the page and removed the `view homepage` section which had previously pointed to the link

![Screenshot from 2023-01-17 14-13-52](https://user-images.githubusercontent.com/51425873/212909399-09585d00-2ce9-45ef-a352-2e2fab7dfab3.png)

- I added the `license`, and  `acknowledgments` pages to the `more` section of the navbar

![Screenshot from 2023-01-17 14-14-32](https://user-images.githubusercontent.com/51425873/212909939-a04bf401-8e7e-4e7d-986a-070427bd6edf.png)

- I added the `installation` page to the side of the `getting started` page and removed the `contents` section

![Screenshot from 2023-01-17 14-15-37](https://user-images.githubusercontent.com/51425873/212910210-bdf2b4d8-56d4-4644-aedf-b1f84b038b09.png)

and also added the acknowledges page to the redirect list. Please let me know if there are any more changes you will like me to do. I will get on them as soon as possible 🙂

DON-BRAN commented on docs/getting_started.rst:
```diff
@@ -37,103 +37,10 @@ Projects using Zarr
-    license
-    View homepage <https://zarr.dev/>
-
-Indices and tables
```
Hi @joshmoore, could it be placed at the `more` section at the top?

joshmoore replied:
I pushed a commit seeing how it looks on the api page.

joshmoore commented:
![Screenshot 2023-01-18 at 18 35 16](https://user-images.githubusercontent.com/88113/213253763-8545bd94-5da5-4412-9b8e-2b945b8c7752.png)

DON-BRAN commented:
> ![Screenshot 2023-01-18 at 18 35 16](https://user-images.githubusercontent.com/88113/213253763-8545bd94-5da5-4412-9b8e-2b945b8c7752.png)

This is a perfect fit 🚀

joshmoore commented:
Perhaps last call for feedback, @zarr-developers/python-core-devs!

joshmoore commented:
Rolling this into the upcoming 2.14. Thanks, @DON-BRAN!
